### ISSUE TITLE
code review: pruning.py, featurizer.py and softmax.py 

### ISSUE THREAD

**Issue by gmichalo** [HoloClean/HoloClean-Legacy-deprecated#38]
code review for pruning.py, featurizer.py and softmax.py 

This code should create a new feature table and the X tensor for pytorch

**Comment by ah89** [comment]
The sqls checked
